line_filter.go

```diff
@@ -0,0 +1,28 @@
+package run
+
+import "context"
+
+// LineFilter allows modifications of individual lines from Output.
+//
+// An explicit "skip" return parameter is required because many bytes library functions
+// return nil to denote empty lines, which should be preserved: https://github.com/golang/go/issues/46415
+type LineFilter func(line []byte) (newLine []byte, skip bool)
+
+// JQFilter creates a LineFilter that executes a JQ query against each line. Errors at
+// runtime get written to output.
+//
+// Refer to https://github.com/itchyny/gojq for the specifics of supported syntax.
+func JQFilter(query string) (LineFilter, error) {
+	jqCode, err := buildJQ(query)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(line []byte) ([]byte, bool) {
+		b, err := execJQ(context.TODO(), jqCode, line)
```

Comment:
Not sure what to do here - maybe `LineFilter` should accept a context?

Reply:
Or maybe the `JQFilter` should accept a context, though that seems weirdly duplicative of providing context to `run.Cmd`

Reply:
It does kind of make sense to provide ctx to `LineFilter` though, since we never know how much is being done in each filter - maybe you want to make API calls in here? 😆

Reply:
Or maybe new API? `FilterContext`?

Reply:
personally, I don't have a use case that would need context yet (I think?)

the example of `FilterContext` feels a bit unintuitive to me

anyway, don't have a strong opinion here just happy to see `jq` support lol

Reply:
I embed the context for the all-output version of the JQ filter (https://github.com/sourcegraph/run/pull/12/commits/b195239fcaba968e578d96d8f9dbae3cb033af42), but for the per-line JQ filter the API doesn't allow an easy way to do so

Reply:
An alternative way to add a context argument without bloating the interface as much is to remove the return values entirely:

```go
type LineFilter func(line []byte) (newLine []byte, skip bool)
```

becomes:

```go
type LineFilter func(ctx context.Context, line []byte, out io.Writer)
```

Another thought: should LineFilter be allowed to return errors? Right now for JQ the errors are written to the output

Reply:
Further discussion: https://github.com/sourcegraph/run/issues/14